contracts/bonding/BondingManager.sol

```diff
@@ -220,12 +220,18 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
 
         uint256 currentRound = roundsManager().currentRound();
 
-        if (delegatorStatus(msg.sender) == DelegatorStatus.Unbonded) {
+        if (delegatorStatus(msg.sender) != DelegatorStatus.Bonded) {
```

Comment:
Previously, we set the delegator's start round only if it was in the unbonded state or if it had a delegate and was delegating to someone else. Now, we set the delegator's start round no matter what unless its already bonded. If it is already bonded, we set its starts only if it is changing its delegation, otherwise we do not set its start round. This means that if you call `bond()` from the unbonding, unbonded or pending state your start round is updated.